### PR TITLE
deps: upgrade deps to get rid of Buffer constructor

### PR DESCRIPTION
**Pull request by aduh95**
**Summary**

Node 10 is about to become LTS and Yarn users are still receiving warnings from the use of the deprecated `Buffer` constructor. This PR updates the dependencies triggering those warnings.

Motivated by https://github.com/yarnpkg/yarn/pull/5934#issuecomment-406346724 (thank you @jacobq).

We've come from a long way to get rid of those warnings in yarn. Here the list of the packages which have been upgraded related to the issue #5477:

- [x] yarn (Fixed by: https://github.com/yarnpkg/yarn/pull/5731)
- [x] tar-fs (Fixed by: https://github.com/yarnpkg/yarn/pull/5752)
- [x] tar-stream (Fixed by: https://github.com/yarnpkg/yarn/pull/5752)
- [x] v8-compile-cache (Fixed by: https://github.com/yarnpkg/yarn/pull/5763)
- [x] natives (Fixed by: https://github.com/yarnpkg/yarn/pull/5763)
- [x] sshpk (Fixed by: https://github.com/yarnpkg/yarn/pull/5934)

The list of dependencies that are fixed by this PR:

- [x] bl (BufferList) (1.2.1 -> 1.2.2)
- [x] buffer-alloc (1.1.0 -> 1.2.0)
- [x] duplexify (3.5.0 -> 3.6.0)
- [x] ecc-jsbn (0.1.1 -> 0.2.0)
- [x] peek-stream (1.1.2 -> 1.1.3)

When running `yarn build-bundle` then searching for all the occurrences of `new Buffer(`, there are now 14 search results:

- 2 are comments
- 7 are compatibility fallback
- 1 is from `buffer-fill`, but the code is not used by any of the `yarn` command
- 3 are from `asn1`
- 1 is from `http-signature`

To sum up, here are the PR we are waiting to merge to be done with this deprecation warning in yarn:

* [ ] asn1 (Refs: https://github.com/joyent/node-asn1/pull/30)
* [ ] http-signature (Refs: https://github.com/joyent/node-http-signature/pull/67)

**Test plan**

CI

### CODE REVIEW

**Comment by aduh95** [comment]
The PR on `asn1` has been merged, I will rebase as soon as they release a new version with the fix.

**Comment by arcanis** [comment]
Great work, thanks! Can you check whether the tests run? It's curious they pass on Node 4 but not the others 😕

**Comment by aduh95** [comment]
I am not sure of what appended there, I am running Node 8 and I have few tests failing on my machine due to timeout: I blame my connection, bet let's see how the CI will say this time.

Now `asn1` has released a new version, I have rebased my branch and update the dependencies. Somehow, there is now a new package in the dependency tree which is using the deprecated `Buffer` constructor...

<details>
```
yarn why v1.10.0-0
[1/4] Why do we have the module "chardet"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "chardet@0.4.2"
info Reasons this module exists
   - "external-editor" depends on it
   - Hoisted from "external-editor#chardet"
info Disk size without dependencies: "108KB"
info Disk size with unique dependencies: "108KB"
info Disk size with transitive dependencies: "108KB"
info Number of shared dependencies: 0
$ bin/yarn why external-editor
=> Found "external-editor@2.2.0"
info Has been hoisted to "external-editor"
info Reasons this module exists
   - Hoisted from "inquirer#external-editor"
   - Hoisted from "commitizen#opencollective#inquirer#external-editor"
info Disk size without dependencies: "72KB"
info Disk size with unique dependencies: "628KB"
info Disk size with transitive dependencies: "704KB"
info Number of shared dependencies: 5
=> Found "commitizen#external-editor@1.1.1"
info Reasons this module exists
   - "commitizen#inquirer" depends on it
   - Hoisted from "commitizen#inquirer#external-editor"
```
</details>

**Comment by aduh95** [comment]
So now the linter [crashes](https://circleci.com/gh/yarnpkg/yarn/19147?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link), I guess some unrelated dependencies have upgraded to version containing breaking changes.

I will froze the related dependencies to the previous used version, another PR should look into that.

File: package.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "babylon": "^6.5.0",
     "commitizen": "^2.9.6",
     "cz-conventional-changelog": "^2.0.0",
-    "eslint": "^4.3.0",
```
Comment (jacobq) [inline package.json]: Shouldn't you be able to keep the `^` for `eslint` and `prettier`? (Shouldn't be any breaking changes unless they are not semver compliant, right?) Does the CI build pass if you just keep the `chalk@2` resolution below?

Comment (aduh95) [reply package.json]: Nope, here what happens when eslint gets an upgrade: https://circleci.com/gh/yarnpkg/yarn/19147?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

And for Prettier, well try it yourself, it reports about 1,400 errors because the defaults evolved. 

As stated in https://github.com/yarnpkg/yarn/pull/6208#issuecomment-410082800, I think those problems should be addressed in a separate PR.

File: package.json [hunk 2]
```diff
@@ -92,12 +92,13 @@
...
     "minimatch": "^3.0.4",
     "mock-stdin": "^0.3.0",
-    "prettier": "^1.5.2",
+    "prettier": "1.5.2",
     "temp": "^0.8.3",
     "webpack": "^2.1.0-beta.25",
     "yargs": "^6.3.0"
   },
   "resolutions": {
+    "chalk@2": "2.3.2",
```
Comment (arcanis) [inline package.json]: Rather than doing this, can you update the lockfile to merge the three `chalk` entries together, since they all seem to accept `2.3.2` as valid resolution (I know there's a "do no edit" warning at the top of the file, but it's actually safe to do if you know what you're doing)?

**Comment by arcanis** [comment]
@imsnif Can you give a look at this PR? I'm not entirely sure I understand why the integrity hashes are changing like crazy by switching from sha-1 to sha-512 and vice versa.

**Comment by imsnif** [comment]
Sure - I'll take a look sometime in the next two days.
